# Single-cell RNA-seq tools
## Pipelines:
  - Seurat: https://satijalab.org/seurat/index.html
    > Tutorial: https://github.com/hbctraining/scRNA-seq_online/tree/master/lessons
  - Scanpy: https://scanpy.readthedocs.io/en/stable/
  - Scvi-tools: https://docs.scvi-tools.org/en/stable/index.html#
  - KNetL(with a new view of dimensionality reduction) from iCellR: https://github.com/rezakj/iCellR
## Automatic cell type indentification:
  - Cortal, A., Martignetti, L., Six, E. et al. Gene signature extraction and cell identity recognition at the single-cell level with Cell-ID. Nat Biotechnol 39, 1095–1102 (2021). https://doi.org/10.1038/s41587-021-00896-6
    > https://github.com/RausellLab/CelliD;
  - Pliner, H.A., Shendure, J. & Trapnell, C. Supervised classification enables rapid annotation of cell atlases. Nat Methods 16, 983–986 (2019). https://doi.org/10.1038/s41592-019-0535-3
    > https://cole-trapnell-lab.github.io/garnett/;
    > https://cole-trapnell-lab.github.io/monocle3/docs/clustering/#garnett
## Cell relationship, fate, velocity and trajectory:
  - https://www.10xgenomics.com/resources/analysis-guides/trajectory-analysis-using-10x-Genomics-single-cell-gene-expression-data
  - Schwartz, G.W., Zhou, Y., Petrovic, J. et al. TooManyCells identifies and visualizes relationships of single-cell clades. Nat Methods 17, 405–413 (2020). https://doi.org/10.1038/s41592-020-0748-5
    > https://github.com/GregorySchwartz/too-many-cells
  - Xiaojie Qiu, Yan Zhang, Jorge D. Martin-Rufino, et al, Jonathan S. Weissman,Mapping transcriptomic vector fields of single cells,Cell,2022,,ISSN 0092-8674,https://doi.org/10.1016/j.cell.2021.12.045.
    > https://github.com/aristoteleo/dynamo-release;
    > https://dynamo-release.readthedocs.io/en/latest/
  - Lange, M., Bergen, V., Klein, M. et al. CellRank for directed single-cell fate mapping. Nat Methods 19, 159–170 (2022). https://doi.org/10.1038/s41592-021-01346-6
    > https://cellrank.readthedocs.io/en/stable/
  - La Manno, G., Soldatov, R., Zeisel, A. et al. RNA velocity of single cells. Nature 560, 494–498 (2018). https://doi.org/10.1038/s41586-018-0414-6
    > http://velocyto.org/
  - Bergen, V., Lange, M., Peidli, S. et al. Generalizing RNA velocity to transient cell states through dynamical modeling. Nat Biotechnol 38, 1408–1414 (2020). https://doi.org/10.1038/s41587-020-0591-3
    > https://scvelo.readthedocs.io/
  - Tedesco, M., Giannese, F., Lazarević, D. et al. Chromatin Velocity reveals epigenetic dynamics by single-cell profiling of heterochromatin and euchromatin. Nat Biotechnol 40, 235–244 (2022). https://doi.org/10.1038/s41587-021-01031-1
    > Chromatin Velocity: https://github.com/leomorelli/scGET; https://github.com/dawe/scatACC
## Cell-cell interaction:
  - Jin, S., Guerrero-Juarez, C.F., Zhang, L. et al. Inference and analysis of cell-cell communication using CellChat. Nat Commun 12, 1088 (2021). https://doi.org/10.1038/s41467-021-21246-9
    > CellChat: http://www.cellchat.org/; https://github.com/sqjin/CellChat
## Differential abundance testing:
  - Dann, E., Henderson, N.C., Teichmann, S.A. et al. Differential abundance testing on single-cell data using k-nearest neighbor graphs. Nat Biotechnol 40, 245–253 (2022). https://doi.org/10.1038/s41587-021-01033-z
    > Milo: https://github.com/MarioniLab/miloR
## Doublet removal:
  - McGinnis, Christopher S., Lyndsay M. Murrow, and Zev J. Gartner. "DoubletFinder: doublet detection in single-cell RNA sequencing data using artificial nearest neighbors." Cell systems 8.4 (2019): 329-337.
    > https://github.com/chris-mcginnis-ucsf/DoubletFinder
  - Wolock, Samuel L., Romain Lopez, and Allon M. Klein. "Scrublet: computational identification of cell doublets in single-cell transcriptomic data." Cell systems 8.4 (2019): 281-291.
    > https://github.com/swolock/scrublet
# Current techniques
![image](https://user-images.githubusercontent.com/19710340/169665170-60d55541-1571-4586-ba30-0a5da50cf89f.png)
![image](https://user-images.githubusercontent.com/19710340/169665211-092743a0-dbe7-4bfc-b983-849f5465afbf.png)
> Ogbeide, Silvia, et al. "Into the multiverse: advances in single-cell multiomic profiling." Trends in Genetics (2022).https://doi.org/10.1016/j.tig.2022.03.015
